### PR TITLE
Add task start and end timestamps

### PR DESCRIPTION
**Pull request by eduardb**
Hello there! 
I am planning to use this plugin for pushing build telemetry into our observability platform. However, in order to be able to build the tasks as a tree of spans, we need the start and end timestamps for them. So I am adding these fields to `MeasuredTask`.
Please let me know if there's any kind of improvement you'd like to see in this PR to make it shippable. Cheers!

### CODE REVIEW

File: measure-builds/src/test/java/com/automattic/android/measure/BuildTimePluginTest.kt [hunk 1]
```diff
@@ -138,6 +138,7 @@ class BuildTimePluginTest {
             assertThat(executionData.executedTasks).hasSize(1).first().satisfies({ task ->
                 assertThat(task.name).isEqualTo(":help")
                 assertThat(task.state).isEqualTo(MeasuredTask.State.EXECUTED)
+                assertThat(task.endTimestamp - task.startTimestamp).isEqualTo(task.duration.inWholeMilliseconds)
```
Comment (eduardb) [inline measure-builds/src/test/java/com/automattic/android/measure/BuildTimePluginTest.kt:141]: Arguably, this assertion shouldn't be part of this test in particular, but it was quite convenient. But I can write a new dedicated test if you insist.

**Review by wzieba — APPROVED** [review]
Thanks for the contribution @eduardb ! The `MeasuredTask#duration` is now functionally dependent from `startTimestamp` and `endTimestamp`, but I think this is a minor issue (or not an issue at all).

**Comment by eduardb** [comment]
> The `MeasuredTask#duration` is now functionally dependent from `startTimestamp` and `endTimestamp`, but I think this is a minor issue (or not an issue at all).

Oh, that's a good point. Perhaps only `startTimestamp` should be added, and the end timestamp can be calculated by whoever needs it by adding `duration` to it? Let me know if you'd like me to quickly change this.

**Comment by eduardb** [comment]
Thanks for merging, btw! ❤️ 
Are you also planning on doing a release with this change? :D

**Comment by eduardb** [comment]
Done here: https://github.com/Automattic/measure-builds-gradle-plugin/pull/51

**Comment by wzieba** [comment]
Thank you for the follow up PR @eduardb . Publishing to the gradle plugin portal was successful, 3.1.0 should soon be available.

**Pull request merged**